pkg/controller/restore_controller.go

```diff
@@ -462,43 +472,95 @@ func mostRecentCompletedBackup(backups []*api.Backup) *api.Backup {
+	if err != nil {
+		if !apierrors.IsNotFound(err) {
+			return backupInfo{}, errors.WithStack(err)
+		}
+
+		logContext := c.logger.WithField("backupName", backupName)
+		logContext.Debug("Backup not found in backupLister, checking each backup location directly, starting with default...")
+		return c.fetchFromBackupStorage(backupName, pluginManager)
 	}
 
-	if !apierrors.IsNotFound(err) {
-		return nil, errors.WithStack(err)
+	location, err := c.backupLocationLister.BackupStorageLocations(c.namespace).Get(info.backup.Spec.StorageLocation)
+	if err != nil {
+		return backupInfo{}, err
 	}
 
-	logContext := c.logger.WithField("backupName", name)
+	info.objectStore, err = getObjectStoreForLocation(location, pluginManager)
+	if err != nil {
+		return backupInfo{}, errors.Wrap(err, "error initializing object store")
+	}
 
-	logContext.Debug("Backup not found in backupLister, checking object storage directly")
-	backup, err = c.getBackup(objectStore, c.bucket, name)
+	return info, nil
+}
+
+// fetchFromBackupStorage checks each backup storage location, starting with the default,
+// looking for a backup that matches the given backup name.
+func (c *restoreController) fetchFromBackupStorage(backupName string, pluginManager plugin.Manager) (backupInfo, error) {
+	locations, err := c.backupLocationLister.BackupStorageLocations(c.namespace).List(labels.Everything())
 	if err != nil {
-		return nil, err
+		return backupInfo{}, err
 	}
 
-	// ResourceVersion needs to be cleared in order to create the object in the API
-	backup.ResourceVersion = ""
-	// Clear out the namespace too, just in case
-	backup.Namespace = ""
+	idx := 1
+	orderedLocations := make([]*api.BackupStorageLocation, len(locations))
+	for i := 0; i < len(locations); i++ {
+		if locations[i].Name == c.defaultBackupLocation {
+			orderedLocations[0] = locations[i]
+		} else {
+			orderedLocations[idx] = locations[i]
+			idx++
+		}
+	}
 
-	created, createErr := c.backupClient.Backups(c.namespace).Create(backup)
-	if createErr != nil {
-		logContext.WithError(errors.WithStack(createErr)).Error("Unable to create API object for Backup")
-	} else {
-		backup = created
+	logContext := c.logger.WithField("backupName", backupName)
+	var info backupInfo
+	var found bool
+	for _, location := range orderedLocations {
+		info.objectStore, err = getObjectStoreForLocation(location, pluginManager)
+		if err != nil {
+			logContext.WithError(errors.WithStack(err)).Error("Unable to get an object store in the location ", location.Name)
+			continue
+		}
+
+		info.backup, err = c.getBackup(info.objectStore, location.Spec.ObjectStorage.Bucket, backupName)
+		if err != nil {
+			logContext.WithError(errors.WithStack(err)).Error("Unable to find a backup in the location ", location.Name)
+			continue
+		}
+
+		// ResourceVersion needs to be cleared in order to create the object in the API
+		info.backup.ResourceVersion = ""
+		// Clear out the namespace too, just in case
```

Comment:
Just in case the backup was made in a different cluster, with a different namespace? That might be worth saying explicitly

Reply:
Will do!